Increase timeout for game assets retrieval to prevent timeouts with large libraries

PadTrick opened the pull request:
### Problem
Users with large game libraries (600+ titles) or slower network connections experience `ReadTimeout` errors when retrieving their game list through `get_game_assets()`. The current 10-second timeout is insufficient for these cases.

### Solution
- Increased timeout from 10 to 60 seconds in `get_game_assets()` method
- This provides sufficient time for the API call to complete even with large libraries

### Testing
- Tested with user having ~600 games in library
- Previously failed with 10s timeout, now works reliably with 60s timeout
- No negative impact for users with smaller libraries

### Notes
The timeout increase only affects the specific API call that retrieves game assets, not other API operations.

**Related:** Heroic-Games-Launcher/HeroicGamesLauncher#4945

PadTrick committed:
fix(api): increase timeout for get_game_assets to 60 seconds

Fixes timeout issues when retrieving game lists for users with large libraries (>600 titles) or slow connections.
The previous 10-second timeout was insufficient for some users, causing ReadTimeout errors.

BananaWorks07 commented on legendary/api/egs.py:
```diff
@@ -167,8 +167,9 @@ def get_external_auths(self):
+        forced_timeout = 60.0
         r = self.session.get(f'https://{self._launcher_host}/launcher/api/public/assets/{platform}',
-                             params=dict(label=label), timeout=self.request_timeout)
+                             params=dict(label=label), timeout=forced_timeout)  # placed 60 second timeout
```
fwiw, we don't need this change at all, setting --api-timeout in the command line options will also work. However, we can definitely increase the default api timeout to lets say 20 seconds which is located around line 33 in this same file. If 20 is not enough, then we might need to introduce a separate timeout variable just for the game assets.

PadTrick replied:
yeah sounds good. 
i didnt wanted to change the default for all calls, because this is the only one that i needed to change.
maybe it is also possible to adjust the timeout with a slider or dropdown in heroic (10,20,30, .. up to 60)?
i'm just happy that i can finally use heroic again :)

PadTrick commented:
**Update:** I've implemented a more elegant solution with a configurable timeout parameter instead of the fixed increase proposed here.

The new approach in #26 adds a `--asset-timeout` flag that:
- Maintains the default 10s timeout for all users
- Allows customizable timeouts only when needed
- Provides better flexibility and doesn't affect users with small libraries

This addresses the feedback from our discussion and is a much cleaner architecture. I'm closing this PR in favor of the new implementation.
